drivers: rtc: rx8130ce: day alarm fix

Mattemagikern opened the pull request:
The day alarm was not being set correctly. The day alarm should be set using the WADA bit in the control register. This patch fixes the issue by setting the WADA bit in the control register when setting the day alarm.

bjarki-andreasen commented on drivers/rtc/Kconfig.rx8130ce:
```diff
@@ -3,9 +3,9 @@
 	select I2C
 	help
-	  Enable RTC driver for the Epson rx8130ce rtc
+	   Enable RTC driver for the Epson RX8130CE RTC
```
Remove the added preceeding space, indent should be two spaces :)

bjarki-andreasen reviewed (commented):
Nice work :) Could you split the PR into two parts, one fixing the bug, one doing the cleanup? The feature freeze limits us to merging bug fixes, I think the cleanup might be fine but I don't know :)

Mattemagikern committed:
drivers: rtc: rx8130ce: day alarm fix

The day alarm was not being set correctly. The day alarm should be set
using the WADA bit in the control register. This patch fixes the issue
by setting the WADA bit in the control register when setting the day alarm.

Signed-off-by: Måns Ansgariusson <Mansgariusson@gmail.com>

Mattemagikern commented:
Hello @bjarki-andreasen, 

This is just the bug fix part :) 

I'll fix another PR for the clean-up code :)

Mattemagikern commented:
on a semi-different topic @bjarki-andreasen: 

Would a patch with added tests for induvidual alarms be appriciated in the `rtc_api/` tests? 
So not an exact time as it is today, but "on-every-hour-on-the-hour" alarm or every-minutes, every-day etc etc. 

They could be selected/skipped depending on the alarm mask that is configured. 

I think that is the only thing missing from the API tests as I've found it today.

bjarki-andreasen commented:
> on a semi-different topic @bjarki-andreasen:
> 
> Would a patch with added tests for induvidual alarms be appriciated in the `rtc_api/` tests? So not an exact time as it is today, but "on-every-hour-on-the-hour" alarm or every-minutes, every-day etc etc.
> 
> They could be selected/skipped depending on the alarm mask that is configured.
> 
> I think that is the only thing missing from the API tests as I've found it today.

It depends, RTC alarms vary wildly in what field combinations, hour, minute, day etc. are usable, originally the test suite did test every hour on the hour if the flag was supported, but some RTCs need all fields set for the alarm to work, they can't do "periodic" alarms matching on only a few fields, hence only "single shot" alarms, setting all fields defined in the `CONFIG_TEST_RTC_ALARM_TIME_MASK`, are tested now since that is the "lowest common demoninator", all RTCs support that :)

Extending the test suite with an additional test, specifically for testing alarms which support the "simple" compare enabled field to time, if all match, trigger, style hardware, enabled with a Kconfig like `TEST_RTC_ALARM_PERIODIC` or something could be added to test it for RTCs which support it. Just please avoid extending `test_alarm.c`, add a new one which names something like `test_alarm_periodic.c` which specifically tests the alarm being triggered, not whether the alarm was set properly for example :)
